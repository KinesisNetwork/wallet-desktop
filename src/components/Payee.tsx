import * as React from 'react'

import { PayeeForm } from '@containers/PayeeForm'
import { PayeeList } from '@containers/PayeeList'

export const Payee: React.SFC = () => (
  <div className='columns'>
    <div className='column is-half'>
      <PayeeList />
    </div>
    <div className='column'>
      <div className='vertical-spaced has-text-centered'>
        <h1 className='title-heading'>ADD A NEW PAYEE</h1>
        <section className='section'>
          <PayeeForm />
        </section>
      </div>
    </div>
  </div>
)

Payee.displayName = 'Payee'
