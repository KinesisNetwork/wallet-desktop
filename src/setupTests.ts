import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
