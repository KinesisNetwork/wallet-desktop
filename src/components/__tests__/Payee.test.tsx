import { shallow } from 'enzyme'
import * as React from 'react'

import { Payee } from '@components/Payee'
import '../../setupTests'

describe('Payee', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Payee />)

    expect(wrapper).toMatchSnapshot()
  })

  it('returns PayeeList and PayeeForm components', () => {
    const wrapper = shallow(<Payee />)

    expect(wrapper.children()).toHaveLength(2)
    expect(wrapper.find('Connect(PayeeList)')).toHaveLength(1)
    expect(wrapper.find('Connect(PayeeForm)')).toHaveLength(1)
    expect(wrapper.find('h1').text()).toEqual('ADD A NEW PAYEE')
  })
})
