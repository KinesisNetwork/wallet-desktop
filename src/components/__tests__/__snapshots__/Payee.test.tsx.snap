// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payee renders correctly 1`] = `
<div
  className="columns"
>
  <div
    className="column is-half"
  >
    <Connect(PayeeList) />
  </div>
  <div
    className="column"
  >
    <div
      className="vertical-spaced has-text-centered"
    >
      <h1
        className="title-heading"
      >
        ADD A NEW PAYEE
      </h1>
      <section
        className="section"
      >
        <Connect(PayeeForm) />
      </section>
    </div>
  </div>
</div>
`;
